CertificateTests/CertificateTest.cs

```diff
@@ -0,0 +1,169 @@
+using AventStack.ExtentReports.Reporter;
+using AventStack.ExtentReports;
+using NunitCompetition.Utilities;
+using NUnit.Framework;
+using NunitCompetition.Page;
+using OpenQA.Selenium;
+
+namespace NunitCompetition.CertificateTests
+{
+
+    [TestFixture]
+    public class Tests : CommonHook
+    {
+        public Tests()
+        {
+            Console.WriteLine("Tests class instance created.");
+        }
+        [OneTimeSetUp]
+        public void EducationTest()
+        {
+            extent.CreateTest("Tests for Certificate Feature");
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            
+            // Create the directory for screenshots
+            Directory.CreateDirectory("Screenshots");
+
+            // Use the test context or JSON file information for the test name
+            string testName = TestContext.CurrentContext.Test.Name; // Default test name
+            string testDescription = TestContext.CurrentContext.Test.Properties.Get("Description") as string;
+
+            // Extract test name from JSON file if applicable (optional improvement)
+            if (TestContext.CurrentContext.Test.Arguments.Length > 0 && TestContext.CurrentContext.Test.Arguments[0] is string jsonFilePath)
+            {
+                testName = Path.GetFileNameWithoutExtension(jsonFilePath); // Use JSON file name as test name
+            }
+
+            // Log the test name and description in the Extent Report
+            test = extent.CreateTest(testName, testDescription);
+            
+
+
+            // Initialize WebDriver
+            NunitCompetition.Utilities.WebDriverManager.InitializeDriver();
+            Login.LoginActions();
+            CertificatePage.GoToTab(driver);
+            CertificatePage.DeleteAllElement(driver);
```

Comment:
delete all should happen in the one time hook, again the reason is if you ran tests in parallel you will see that tests fail due to other reasons! again to do with how hooks are executed in what order